src/v0/sources/shopify/shopify.util.test.js

```diff
@@ -65,20 +59,71 @@ describe('Shopify Utils Test', () => {
     });
   });
 
-  describe('set AnonymousId without using Redis Test Cases', () => {
-    it('Properties containing cartToken', () => {
+  describe('Check for valid cart update event test cases', () => {
+    const { checkAndUpdateCartItems } = require('./util');
+    it('Event containing token and nothing is retreived from redis', async () => {
+      const input = {
+        token: 'token_not_in_redis',
+        line_items: [
+          {
+            prod_id: 'prod_1',
+            quantity: 1,
+          },
+        ],
+      };
+      const expectedOutput = true;
+      const output = await checkAndUpdateCartItems(input);
+      expect(output).toEqual(expectedOutput);
+    });
+
+    it('Event contain id for cart_update event and isValid', async () => {
+      const input = {
+        id: 'shopify_test2',
+        line_items: [
+          {
+            prod_id: 'prod_1',
+            quantity: 1,
+          },
+        ],
+      };
+
+      const expectedOutput = true;
+      const output = await checkAndUpdateCartItems(input);
+      expect(output).toEqual(expectedOutput);
+    });
+
+    it('Event contain id for cart_update event and isInValid', async () => {
+      const input = {
+        id: 'shopify_test_duplicate_cart',
+        line_items: [
+          {
+            prod_id: 'prod_1',
+            quantity: 1,
+          },
+        ],
+      };
+
+      const expectedOutput = false;
+      const output = await checkAndUpdateCartItems(input);
+      expect(output).toEqual(expectedOutput);
+    });
+  });
+
+  describe(' Test Cases -> set AnonymousId without using Redis', () => {
+    const { getAnonymousId } = require('./util');
```

Comment:
move imports outside describe

Reply:
Same as above but Let me try if something can be worked here

Reply:
Done